res/controllers/common-hid-packet-parser.js

```diff
@@ -1413,33 +1414,11 @@ class HIDController {
      *     can't be found.
      */
     getOutputField(m_group, m_name) {
-        for (const packet_name in this.OutputPackets) {
-            const packet = this.OutputPackets[packet_name];
-            for (const group_name in packet.groups) {
-                const group = packet.groups[group_name];
-                for (const field_name in group) {
-                    const field = group[field_name];
-                    if (field.type === "bitvector") {
-                        for (const bit_id in field.value.bits) {
-                            const bit = field.value.bits[bit_id];
-                            if (bit.mapped_group === m_group && bit.mapped_name === m_name) {
-                                return bit;
-                            }
-                            if (bit.group === m_group && bit.name === m_name) {
-                                return bit;
-                            }
-                        }
-                        continue;
-                    }
-                    if (field.mapped_group === m_group && field.mapped_name === m_name) {
-                        return field;
-                    }
-                    if (field.group === m_group && field.name === m_name) {
-                        return field;
-                    }
-                }
-            }
+        const field = this.OutputFieldLookup.get([m_group, m_name].toString());
+        if (field) {
+            return field;
         }
+
         return undefined;
```

Comment:
Are you sure that there won't be problem with colliding keys when storing everything in the `Map`?

Reply:
Is the `@todo` at the top of the method now obsolete?

Reply:
I came to the conclusion, that this can handle all cases, which the old code could handle. This is because the old code had a return statement after all four if cases for detecting a value pair. Therefore only the first cases matters, and this is the value stored in the map now.
Please cross-check my assumption.

Reply:
I reread your code and I also came to the conclusion that in case collisions occur, they will still be handled the same as the previous code. Though imo that is only because you encoded the precedence present in the removed search code into the code that fills the lookup map.

Also my suggestion in the first comment still applies, or do you have thoughts in why that simplification would be incorrect?

Reply:
Your suggestion of the first commit makes sense to me: map.prototype.get() returns undefined, in case that the key doesn't exist in the map. I will test this in practice!

Reply:
The only difference in code I could see that it would behave differently if there was a falsy value (other than undefined) being stored in the map, but since that is not the case, the two things should be equivalent.